### PR TITLE
[ci] Move OTBN sim tests to nightly

### PR DESCRIPTION
**Pull request by jwnrt**
Experimental PR to move OTBN simulator tests to the nightly CI as these need a long running time.

This moves the tests found in a certain BUILD file, not specifically those which take a long time or which use the OTBN simulator.

I would like a more robust way of deciding which tests run in CI and which run in nightly, but this is beyond my Bazel and CI knowledge.

### CODE REVIEW

**Review by rswarbrick — APPROVED** [review]
It would be really nice if we could (one day) move this to be "opt in" to enable things for CI, instead of "opt out", which is what this does. But it definitely looks sensible for now, at least!

Thanks!

**Comment by pamaury** [comment]
The problem with opt in is that there are hundreds of tests, so manually tagging them is tedious and error prone. I agree though that the current approach to opt-out (ie move to nightly) is not ideal. We could have a `nightly` tag so that the CI filters out the nightly tests and the nightly only runs the nightly test.

**Comment by jwnrt** [comment]
> The problem with opt in is that there are hundreds of tests, so manually tagging them is tedious and error prone.

Do you know of any way to opt-in/out (or apply tags) for parts of the tree instead of individual tests? e.g. you opt in `//sw/device` as a whole, and manually tag tests as nightly where required?

**Comment by milesdai** [comment]
Oops I missed this PR. I've opened #19619 to move the OTBN sim tests into a separate job to run in parallel with the SW Build and Test job to unblock CI. So far, it seems to be working okay, but if we overload the Azure runners and start seeing high queuing, then we might want to move it to nightly.

From the CI run of that PR, the sim tests are all pretty quick with the exception of `//sw/otbn/crypto/tests:rsa_keygen_checkpq_test` which takes about 30-40 min depending on what machine you run it on. The timeout is in fact set to `eternal` which is one option to filter on. We would want to make sure we've standardized on using the timeouts this way across the repo though to avoid unintentionally sending tests to nightly. Another option is to construct a bazel query to do this filtering.

**Comment by jadephilipoom** [comment]
For my selfish purposes it would be nice to not move _all_ OTBN simulator tests to nightly; they do sometimes catch errors I've introduced in OTBN code before merging. For example, if I change something in p256, there are 10 tests that could potentially be affected and I might not have run all of them locally. Either special-casing `rsa_keygen_checkpq`, filtering based on `timeout=eternal`, or adding some additional identifier like a `nightly` tag or something would be completely fine with me.

**Comment by jwnrt** [comment]
> it would be nice to not move all OTBN simulator tests to nightly

That’s absolutely fine, I don’t intend for this to go through with everything moved to nightly.

> The timeout is in fact set to eternal which is one option to filter on. We would want to make sure we've standardized on using the timeouts this way across the repo though to avoid unintentionally sending tests to nightly. Another option is to construct a bazel query to do this filtering.

Yeah, I think a tag for “nightly” might be good. As @rswarbrick says, maybe reversing the tags for CI / nightly to make CI opt-in would be more thoughtful in the future, but for now given the proportions I think tagging nightly tests and filtering then makes sense.

**Comment by jwnrt** [comment]
This is being superseded by https://github.com/lowRISC/opentitan/pull/20141 and other follow ups.

**Pull request closed without merging**
